buildSrc/subprojects/basics/src/main/kotlin/gradlebuild/minify.gradle.kts

```diff
@@ -52,7 +52,7 @@ plugins.withId("java-base") {
     afterEvaluate {
         configurations.all {
             // everywhere where we resolve, prefer the minified version
-            if (isCanBeResolved && !isCanBeConsumed) {
+            if (isCanBeResolved && !isCanBeConsumed && this.name != "currentClasspath") {
```

Comment:
There has to be a better way to resolve this. The problem I'm working around here is that binary compatibility plugin needs to select only configurations with `attribute(GradleModuleApiAttribute.attribute as Attribute<GradleModuleApiAttribute>, GradleModuleApiAttribute.API)` in `currentClasspath` configuration.
Here, however, a minified attribute selector is injected (by the way, isn't this the kind of cross-configuration that we want to avoid?) which results in everything that does not match my required attribute, match minified attribute instead and get selected.

Reply:
In other words, here's what happens with minified attribute applied: https://ge.gradle.org/s/qyz6j5beka3ci/dependencies?focusedDependency=WzAsMTgsMTIzMyxbMCwxOCxbMTIzMiwxMjMzXV1d&toggled=W1swXSxbMCwxOF0sWzAsMTgsWzEyMzJdXV0
And here's what I want: https://ge.gradle.org/s/utrg7zwotkuu2/dependencies?focusedDependency=WzAsMTgsMTY4NCxbMCwxOCxbMTIzMCwxNjg0XV1d&focusedDependencyView=dependencies_or_failure&toggled=W1swXSxbMCwxOF0sWzAsMTgsWzEyMzBdXV0

Reply:
I think the approach here to use `configurations.all { }` is wrong to begin with. So yes, it is the kind of cross-configuration we want to avoid.

We haven't really figured out how to do this right and it seems we miss something in Gradle. Related:
- https://github.com/gradle/gradle/issues/11831
- https://github.com/gradle/gradle/pull/12089

So in the context of this PR, this workaround is fine I would say. We have to tackle the issue here separately.